Firebase Emulators Lab

ericwindmill opened the pull request:
Adds start and complete flutter projects for Emulators code lab.

## Pre-launch Checklist

- [x] I read the [Effective Dart: Style] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.

[Effective Dart: Style]: https://dart.dev/guides/language/effective-dart/style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat

domesticmouse commented:
Lot of red in the CI pipeline

domesticmouse commented:
Regarding generated files, I leave in generated files that aren't explicitly covered by the `.gitignore` rules from the templates.

ericwindmill commented:
> Regarding generated files, I leave in generated files that aren't explicitly covered by the `.gitignore` rules from the templates.

If you're referring to my commit message, the generated files were from Firebase in the directory for the code lab we removed yesterday, and for some reason just the generated files were sticking around when I changed branches. I don't know enough about git to know why or how

domesticmouse commented on firebase-emulator-suite/complete/lib/firebase_options.dart:
```diff
@@ -0,0 +1,75 @@
+    if (kIsWeb) {
+      return web;
+    }
+    // ignore: missing_enum_constant_in_switch
```
use a `default` case instead of ignoring a linter warning?

ericwindmill replied:
This file is generated by the FlutterFire CLI, so it'll be be overwritten on the users end anyway

domesticmouse replied:
okidat

domesticmouse commented on firebase-emulator-suite/complete/lib/journal_entry_form.dart:
```diff
@@ -0,0 +1,107 @@
+  // Create a global key that uniquely identifies the Form widget
+  // and allows validation of the form.
+  //
+  // Note: This is a GlobalKey<FormState>,
```
I tend to migrate any comments in the code into the body of the codelab.

domesticmouse commented:
@ericwindmill This looks good to me, are you ready to land it?

ericwindmill commented:
@domesticmouse I'm ready. 

Do you merge everything in this repo? That's my guess based on the past, but also feel free to tell me otherwise. It's your repo, so whatever works for you is fine

domesticmouse commented:
> @domesticmouse I'm ready.
> 
> Do you merge everything in this repo? That's my guess based on the past, but also feel free to tell me otherwise. It's your repo, so whatever works for you is fine

Doesn't worry me who merges. I just don't want to merge something before it is ready, but on the flipside I don't want PRs sitting by idly. So I ask what the status is =)
